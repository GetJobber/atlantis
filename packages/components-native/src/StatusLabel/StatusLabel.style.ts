import { StyleSheet } from "react-native";
import { tokens } from "../utils/design";

const statusLabelIconDiameter = tokens["space-base"] - tokens["space-smaller"]; //12px

const indicatorOffset = tokens["space-smallest"] + tokens["space-minuscule"];

// Needs to be hardcoded to 3 as this shouldn't change with the tokens
const statusLabelRadius = 3;

export const styles = StyleSheet.create({
  statusLabelRow: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flexWrap: "nowrap",
  },
  statusLabelText: {
    flexShrink: 1,
  },
  statusLabelIcon: {
    borderRadius: statusLabelRadius,
    backgroundColor: tokens["color-success"],
    width: statusLabelIconDiameter,
    height: statusLabelIconDiameter,
    marginTop: indicatorOffset,
  },
  labelTextStartAligned: {
    flexDirection: "row-reverse",
  },
  innerPad: {
    width: tokens["space-small"],
  },
});
