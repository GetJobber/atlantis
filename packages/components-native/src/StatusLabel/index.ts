export { StatusLabel } from "./StatusLabel";
export type { StatusType, StatusLabelType } from "./StatusLabel";
