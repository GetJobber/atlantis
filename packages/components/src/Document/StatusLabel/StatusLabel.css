.statusWrapper {
  flex: 1 1 auto;
  margin-left: 1rem;
}
