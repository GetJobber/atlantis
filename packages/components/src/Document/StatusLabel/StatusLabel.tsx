import React from "react";
import { InlineLabel } from "../../InlineLabel";
import styles from "./StatusLabel.css";

interface StatusLabelProps {
  readonly status: "draft" | "pending" | "paid";
}

function getStatusLabelColour(status: string) {
  switch (status) {
    case "pending":
      return "yellow";
    case "paid":
      return "green";
    default:
      return "greyBlue";
  }
}

export function StatusLabel({ status }: StatusLabelProps) {
  const statusColour = getStatusLabelColour(status);
  return (
    <span className={styles.statusWrapper}>
      <InlineLabel color={statusColour}>{status}</InlineLabel>
    </span>
  );
}
