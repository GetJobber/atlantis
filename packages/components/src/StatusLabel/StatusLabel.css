:root {
  --statusLabel-icon-diameter: 10px;
}

.statusLabelRow {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  gap: var(--space-small);
  width: fit-content;
}

.labelTextEndAligned {
  flex-direction: row-reverse;
}
