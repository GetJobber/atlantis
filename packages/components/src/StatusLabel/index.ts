export { StatusLabel } from "./StatusLabel";
export type { StatusType } from "./StatusType";
