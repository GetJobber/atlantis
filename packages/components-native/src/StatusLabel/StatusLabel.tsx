import React from "react";
import { View } from "react-native";
import { useStyles } from "./StatusLabel.style";
import { Text } from "../Text";
import { useAtlantisTheme } from "../AtlantisThemeContext";

export type StatusType =
  | "success"
  | "warning"
  | "critical"
  | "inactive"
  | "informative";

export interface StatusLabelType {
  readonly statusLabel: string;
  readonly statusType?: StatusType;
}

interface StatusLabelProps {
  /**
   * Text to display.
   */
  readonly text: string;

  /**
   * Alignment of text
   */
  readonly alignment?: "start" | "end";

  /**
   * Status color of the square beside text
   */
  readonly status?: StatusType;
}

export function StatusLabel({
  text,
  alignment = "end",
  status = "success",
}: StatusLabelProps): JSX.Element {
  const styles = useStyles();

  return (
    <View
      style={[
        styles.statusLabelRow,
        alignment === "start" && styles.labelTextStartAligned,
      ]}
    >
      <View style={styles.statusLabelText}>
        <Text align={alignment} level="textSupporting" variation="subdued">
          {text}
        </Text>
      </View>
      <View style={styles.innerPad} />
      <StatusLabelIcon status={status} styles={styles} />
    </View>
  );
}

interface StatusLabelIconProps {
  readonly status: StatusType;
  readonly styles: ReturnType<typeof useStyles>;
}

function StatusLabelIcon({ status, styles }: StatusLabelIconProps) {
  const { tokens } = useAtlantisTheme();

  return (
    <View
      testID={`${status}LabelIcon`}
      style={[
        styles.statusLabelIcon,
        { backgroundColor: tokens[`color-${status}`] },
      ]}
    />
  );
}
