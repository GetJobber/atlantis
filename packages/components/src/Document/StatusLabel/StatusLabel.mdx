---
menu: Components
---

import { Playground, Props } from "docz";
import { StatusLabel } from "./StatusLabel";

# Status Label

Status labels are used to indicate the current state of a `<Document />`.

## Basic Usage

<Playground>
  <StatusLabel status="draft" />
  <StatusLabel status="pending" />
  <StatusLabel status="paid" />
</Playground>

## Properties

<Props of={StatusLabel} />
