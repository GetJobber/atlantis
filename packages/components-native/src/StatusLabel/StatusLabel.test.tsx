import React from "react";
import { render } from "@testing-library/react-native";
import { StatusLabel } from "./StatusLabel";

describe("StatusLabel", () => {
  describe("alignment", () => {
    describe('when alignment prop set to default ("start")', () => {
      it("should match snapshot", () => {
        const view = render(<StatusLabel text="Start Aligned" />).toJSON();
        expect(view).toMatchSnapshot();
      });
    });

    describe('when alignment prop set to "end"', () => {
      it("should match snapshot", () => {
        const view = render(
          <StatusLabel text="End Aligned" alignment="end" />,
        ).toJSON();
        expect(view).toMatchSnapshot();
      });
    });
  });

  describe("status", () => {
    describe('when status prop set to default ("success")', () => {
      it("should match snapshot", () => {
        const view = render(<StatusLabel text="Success" />).toJSON();
        expect(view).toMatchSnapshot();
      });
    });

    describe('when status prop set to "warning"', () => {
      it("should match snapshot", () => {
        const view = render(
          <StatusLabel text="Warning" status="warning" />,
        ).toJSON();
        expect(view).toMatchSnapshot();
      });
    });

    describe('when status prop set to "critical"', () => {
      it("should match snapshot", () => {
        const view = render(
          <StatusLabel text="Critical" status="critical" />,
        ).toJSON();
        expect(view).toMatchSnapshot();
      });
    });

    describe('when status prop set to "inactive"', () => {
      it("should match snapshot", () => {
        const view = render(
          <StatusLabel text="Inactive" status="inactive" />,
        ).toJSON();
        expect(view).toMatchSnapshot();
      });
    });

    describe('when status prop set to "informative"', () => {
      it("should match snapshot", () => {
        const view = render(
          <StatusLabel text="Informative" status="informative" />,
        ).toJSON();
        expect(view).toMatchSnapshot();
      });
    });
  });
});
