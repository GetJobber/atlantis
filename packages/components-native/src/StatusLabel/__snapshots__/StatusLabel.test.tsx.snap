// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel alignment when alignment prop set to "end" should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      End Aligned
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(125, 176, 14)",
        },
      ]
    }
    testID="successLabelIcon"
  />
</View>
`;

exports[`StatusLabel alignment when alignment prop set to default ("start") should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      Start Aligned
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(125, 176, 14)",
        },
      ]
    }
    testID="successLabelIcon"
  />
</View>
`;

exports[`StatusLabel status when status prop set to "critical" should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      Critical
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(201, 66, 33)",
        },
      ]
    }
    testID="criticalLabelIcon"
  />
</View>
`;

exports[`StatusLabel status when status prop set to "inactive" should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      Inactive
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(77, 105, 116)",
        },
      ]
    }
    testID="inactiveLabelIcon"
  />
</View>
`;

exports[`StatusLabel status when status prop set to "informative" should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      Informative
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(60, 162, 224)",
        },
      ]
    }
    testID="informativeLabelIcon"
  />
</View>
`;

exports[`StatusLabel status when status prop set to "warning" should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      Warning
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(221, 195, 15)",
        },
      ]
    }
    testID="warningLabelIcon"
  />
</View>
`;

exports[`StatusLabel status when status prop set to default ("success") should match snapshot 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "flex-end",
      },
      false,
    ]
  }
>
  <View
    style={
      {
        "flexShrink": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      collapsable={false}
      maxFontSizeMultiplier={1.1428571428571428}
      selectable={true}
      selectionColor="rgb(132, 234, 0)"
      style={
        [
          {
            "fontFamily": "inter-medium",
          },
          {
            "color": "rgba(77, 105, 116, 1)",
          },
          {
            "textAlign": "right",
          },
          {
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "letterSpacing": 0,
          },
        ]
      }
    >
      Success
    </Text>
  </View>
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "rgb(125, 176, 14)",
          "borderRadius": 2,
          "height": 12,
          "marginTop": 3,
          "width": 12,
        },
        {
          "backgroundColor": "rgb(125, 176, 14)",
        },
      ]
    }
    testID="successLabelIcon"
  />
</View>
`;
